sys/include/dbgpin.h

```diff
@@ -0,0 +1,129 @@
+/*
+ * Copyright (C) 2020 Freie Universität Berlin
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser
+ * General Public License v2.1. See the file LICENSE in the top level
+ * directory for more details.
+ */
+
+/**
+ * @defgroup    sys_dbgpin Direct pin control for debugging/profiling
+ * @ingroup     sys
+ *
+ * @warning     This module does not verify the given pin number, so make sure
+ *              the pin numbers you use are actually configured!
+ *
+ * @{
+ * @file
+ * @brief       GPIO wrapper for debugging/profiling purposes
+ *
+ * @author      Hauke Petersen <hauke.petersen@fu-berlin.de>
+ */
+
+#ifndef DBGPIN_H
+#define DBGPIN_H
+
+#include "kernel_defines.h"
+#include "periph/gpio.h"
+
+#ifdef __cplusplus
+extern "C"
+{
+#endif
+
+#ifndef DBGPIN_PINS
+#error Please specify the pins to use with the dbgpin module (DBGPIN_PINS)
+#endif
+
+/**
+ * @brief   Set the given debug pin to HIGH
+ *
+ * @param[in] pin       pin to set, pin number is offset to the list of defined
+ *                      debug pins in DBGPIN_PINS
+ */
+static inline void dbgpin_set(unsigned pin)
+{
+    static const gpio_t dbgpin_pins[] = { DBGPIN_PINS };
+    gpio_set(dbgpin_pins[pin]);
+}
+
+/**
+ * @brief   Set the given debug pin to LOW
+ *
+ * @param[in] pin       pin to set, pin number is offset to the list of defined
+ *                      debug pins in DBGPIN_PINS
+ */
+static inline void dbgpin_clear(unsigned pin)
+{
+    static const gpio_t dbgpin_pins[] = { DBGPIN_PINS };
+    gpio_clear(dbgpin_pins[pin]);
+}
+
+/**
+ * @brief   Toggle the given debug pin
+ *
+ * @param[in] pin       pin to set, pin number is offset to the list of defined
+ *                      debug pins in DBGPIN_PINS
+ */
+static inline void dbgpin_toggle(unsigned pin)
+{
+    static const gpio_t dbgpin_pins[] = { DBGPIN_PINS };
+    gpio_toggle(dbgpin_pins[pin]);
+}
+
+/**
+ * @brief   Output a pulse (2 time toggle) on the given debug pin
+ *
+ * @param[in] pin       pin to set, pin number is offset to the list of defined
+ *                      debug pins in DBGPIN_PINS
+ */
+static inline void dbgpin_pulse(unsigned pin)
+{
+    dbgpin_toggle(pin);
+    dbgpin_toggle(pin);
```

Comment:
```suggestion
    dbgpin_siganl(pin, 2);
```

Reply:
looks nicer indeed, but would lead to an eternal loop as `dbgpin_signal()` is internall calling `dbgpin_pulse()` again, right? So in either of these two we have to call toggle twice I guess

Reply:
Indeed, I didn't look closely enough :-)